homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -30,26 +32,28 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the tankerkoenig sensors."""
-
     coordinator: TankerkoenigDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-    stations = coordinator.stations.values()
     entities = []
-    for station in stations:
-        for fuel in coordinator.fuel_types:
-            if fuel not in station:
-                _LOGGER.warning(
-                    "Station %s does not offer %s fuel", station["id"], fuel
+    for station in coordinator.stations.values():
+        for fuel in (GasType.E10, GasType.E5, GasType.DIESEL):
+            if getattr(station, fuel) is None:
+                name = f"{station.brand} {station.name}"
```

Comment:
since this is not reused, we could also use `f"{station.brand} {station.name}"` directly in the logger message